### PR TITLE
Use explicit imports for the Graph TS library and generated types

### LINKED ISSUE #116: Switch from implicit to explicit (type) imports in mappings
Steps:
- [x] Generate `export`ed types
- [ ] Don't combine runtime/mapping files into one
- [ ] Rename runtime module to `graph` and export all it should export
- [ ] Update the example subgraph to use explicit imports
- [ ] Get things to build and run
- [ ] Generate separate type files for each ABI

### PR DESCRIPTION
**Pull request by Jannis**
Resolves #116.

This PR removes the runtime source code—it now lives in [graph-ts](https://github.com/graphprotocol/graph-ts)—and introduces the following changes:

1. `graph build` and `graph deploy` no longer include type generation. It's too messy.
2. Adds `import { SmartContract, ... } from '@graphprotocol/graph-ts` to the TypeScript files generated from ABIs.
3. No longer builds in a temporary, so the dependencies of the subgraph can be managed with npm/yarn in `node_modules/`.
4. Moves `src/cli/` up one level to `src/` since `src/runtime/` is now gone anyway.
5. Updates the README with new instructions.

Note: This depends on [graph-ts](https://github.com/graphprotocol/graph-ts) being released.

### CODE REVIEW

**Review by leoyvens — COMMENTED** [review]
Epic refactor!

> graph build and graph deploy no longer include type generation.

I remember this being a big deal for UX, could you give more detail on why it's no longer viable?

File: src/codegen.js [hunk 1]
```diff
@@ -280,6 +283,30 @@ class MaybeType {
...
 
+class ModuleImports {
+  constructor(nameOrNames, module) {
+    this.nameOrNames = nameOrNames
+    this.module = module
+  }
+
+  toString() {
+    return `import { ${
+      typeof this.nameOrNames === 'string' ? this.nameOrNames : this.nameOrNames.join(',')
```
[the remark this answers is not in the document]
Comment (Jannis) [reply src/codegen.js]: If `nameOrNames` is a single string, we generate
```typescript
import { foo } from 'bar'
```
If it is an array then we generate a CSV list of the names:
```typescript
import { foo, bar, baz } from 'ruux'
```

**Comment by Jannis** [comment]
@leodasvacas The thing with code/type generation is that you don't really expect the types you rely on to change during build. So if you run codegen, you get some types generated, you build against those types, then you make some changes to e.g. the ABIs and suddenly the build is no longer in sync with the types you had generated before. It's confusing.

I think a two-step process is better:

1. Generate types.
2. Build against those (with `build` or `deploy` command and watch mode).

Then, if you change something that affects the types, go through step 1 and 2 again.

File: src/codegen.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (leodasvacas) [reply src/codegen.js]: Ah ok, I was reading the precedence of `typeof` wrong.

**Review by leoyvens — DISMISSED** [review]
Code looks good, though we should wait for the release of `graph-ts` so we can test this end to end.

**Comment by Jannis** [comment]
Yep, I'll perform a first release in the morning.

**Comment by Jannis** [comment]
@leodasvacas I've made a release for graph-ts, updated the AdChain subgraph (see https://github.com/graphprotocol/adchain-subgraph/pull/14) and have verified that it works. I'm going to merge this and cut a new release.

**Pull request merged**